README.md

```diff
@@ -24,3 +24,106 @@ The code in this repository can be built and tested using the Makefile.
 For more information, see `make help`.
 
 [contrib]: https://github.com/projectcalico/calico/blob/master/CONTRIBUTING_CODE.md
+
+## How can I run Calico Node tests?
+
+Calico Node uses Python and [Nose](https://nose.readthedocs.io/en/latest/) for running tests.
+
+Calico Node tests are divided into the following categories: 
+- `fv`: ???
+- `st`: System integration tests 
+- `k8s-test`: Kubernetes integration tests 
+
+To run any of the above categories use: 
+```
+make <category>
+```
+
+Note, to run the `k8s-test` test category, you will need to have `kubectl` setup on your machine. Go here for [instructions on setting up `kubectl` for your environment](https://kubernetes.io/docs/tasks/tools/install-kubectl/).  
+
+## How can I run a subset of the tests?
+
+If you want to run tests for a specific package for more iterative development, you can filter down into a subset of tests using the following parameters: 
+- For filtering `st` tests, use `ST_TO_RUN` 
+- For filtering `k8s-test` tests, use `K8ST_TO_RUN` 
+
+For example, the following only runs tests within the `bgp` subfolder of the `st` category: 
+```
+make st ST_TO_RUN="tests/st/bgp/"
+```
+
+To only run tests from a single file (e.g. `test_bgp.py`), use the following: 
+```
+make st ST_TO_RUN="tests/st/bgp/test_bgp.py"
+```
+
+To only run a single test within a test file use the below syntax: 
+```
+make st ST_TO_RUN="tests/st/bgp/test_bgp.py:TestReadiness.test_readiness_multihost"
+```
+
+The above examples should apply in the same fashion if you are using `K8ST_TO_RUN` instead for the `k8s-test` category. 
+
+## How do I debug tests?
+There are a number of possible avenues you can use to debug failing tests. 
+
+1. Review the diagnostic logs after the tests finish running 
+- These only show for failed tests
+- Be warned the logs are quite verbose 
+2. Use the parameter `DEBUG_FAILURES` with the Makefile 
+```
+make st DEBUG_FAILURES=true
+```
+- This only applies to `st` tests 
+- A subset of the `st` are wrapped by `debug_failures(fn)` function found in `./tests/st/utils/utils.py`
+- You should be able to wrap whatever test you want 
+- Uses Python's `pdb.set_trace()` library function, allows you to halt executino and step into the containers involved in the test for debugging 
+3. Use manual breakpoints 
+- A more primitive approach is just to add your own breakpoints (using something like `time.sleep(x)`)
+- You should know where to add these after reviewing the diagnostic logs for failed tests (by looking at the stacktraces)
+
+## Linux Dependencies 
+Below is a listing of dependencies that Calico node has on the underlying kernel. The list is currently _partially_ complete. 
```

Comment:
I'd tweak the wording here subtly - these are userspace tools packaged into the node container. Each of these userspace tools does however imply certain underlying kernel support.